### PR TITLE
chore: BED-4141 - added parameters and supporting schemas to openapi src

### PR DESCRIPTION
**Pull request by codydbentley**
## Description

This PR adds common parameters and supporting schemas to the openapi document source.

## Motivation and Context

Part of an ongoing effort to improve API documentation

## How Has This Been Tested?

Manual testing of openapi generation. Documentation only changes

## Types of changes

-   [x] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: packages/go/openapi/src/parameters/header.prefer.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Prefer
+description: Prefer header, used to specify a custom timeout in seconds using the
+  wait parameter as per RFC7240. The default timeout is 30 seconds and maximum timeout
+  is 60 seconds
```
Comment (urangel) [inline packages/go/openapi/src/parameters/header.prefer.yaml:20]: Is this still accurate? I see an unused `defaultTimeout` var in middleware.go

**Review by urangel — APPROVED** [review]
I appreciate that the files are small and easy to read through. It looks like it'll be good for composability too. Other than the prefer header default timeout, the content looks to be accurate to me.

Will there be a `paths` directory under `src` eventually? Is there a reference or documentation that you would suggest reading through to understand the intended structure and organization of this project?

File: packages/go/openapi/src/parameters/header.prefer.yaml (hunk 1 above)
Comment (codydbentley) [reply packages/go/openapi/src/parameters/header.prefer.yaml:20]: That's a good callout, with recent changes to timeouts, this may need an update

**Comment by codydbentley** [comment]
> I appreciate that the files are small and easy to read through. It looks like it'll be good for composability too. Other than the prefer header default timeout, the content looks to be accurate to me.
> 
> Will there be a `paths` directory under `src` eventually? Is there a reference or documentation that you would suggest reading through to understand the intended structure and organization of this project?

@urangel yes, there will be a paths folder. The goal is to produce a single JSON file following this spec: https://spec.openapis.org/oas/v3.0.3. But as for how the source will be broken up into multiple files, there's not really a one-size-fits-all guideline for it. I'm putting together a structure that should be plugin friendly for both JetBrains and VSCode, as well as compatible with tools like Redocly, which is what will be used to compile the yaml source files into a single JSON doc

File: packages/go/openapi/src/parameters/header.prefer.yaml (hunk 1 above)
Comment (codydbentley) [reply packages/go/openapi/src/parameters/header.prefer.yaml]: fixed!

**Review by superlinkx — APPROVED** [review]
LGTM

**Pull request merged**